pyproject.toml

```diff
@@ -0,0 +1,39 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["ethstaker_deposit"]
+
+[project]
+name = "ethstaker_deposit"
+version = "0.1.3"
+description = "Secure key generation for deposits"
+readme = "README.md"
+dependencies = [
+    "click >=7.1.2,<8",
+    "eth-typing",
+    "eth-utils",
+    "pycryptodome",
+    "py-ecc",
+    "ssz"
+]
+requires-python = ">=3.9,<4"
+
+[project.optional-dependencies]
+test = [
```

Comment:
If I remove those, I get errors during the test. I added them because pytest needs them and without I have issues.

Reply:
It could be something else. Here is what I did twice to check on this:

```
python3 -m venv .venv2
source .venv2/bin/activate
pip install click==7.1.2
pip install eth-typing eth-utils pycryptodome py-ecc ssz flake8 jsonschema mypy pytest pytest-asyncio
```

```
pip freeze
```
returned
```
attrs==24.2.0
cached-property==1.5.2
click==7.1.2
cytoolz==0.12.3
eth-hash==0.7.0
eth-typing==5.0.0
eth-utils==5.0.0
flake8==7.1.1
hexbytes==1.2.1
iniconfig==2.0.0
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
lru-dict==1.3.0
mccabe==0.7.0
mypy==1.11.2
mypy-extensions==1.0.0
packaging==24.1
pluggy==1.5.0
py-ecc==7.0.1
pycodestyle==2.12.1
pycryptodome==3.20.0
pyflakes==3.2.0
pyrsistent==0.16.1
pytest==8.3.2
pytest-asyncio==0.24.0
referencing==0.35.1
rpds-py==0.20.0
six==1.16.0
ssz==0.5.0
toolz==0.12.1
typing_extensions==4.12.2
```
```
pytest tests
```
returned
```
platform linux -- Python 3.12.3, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/remyroy/Projects/ethstaker-deposit-cli
configfile: pytest.ini
plugins: asyncio-0.24.0
asyncio: mode=Mode.STRICT, default_loop_scope=function
collected 859 items

tests/test_cli/test_existing_mnemonic.py ......                                                  [  0%]
tests/test_cli/test_exit_transaction_keystore.py .....                                           [  1%]
tests/test_cli/test_exit_transaction_mnemonic.py .......                                         [  2%]
tests/test_cli/test_generate_bls_to_execution_change.py ...                                      [  2%]
tests/test_cli/test_new_mnemonic.py .........                                                    [  3%]
tests/test_cli/test_regeneration.py .                                                            [  3%]
tests/test_credentials.py .                                                                      [  3%]
tests/test_deposit.py ........                                                                   [  4%]
tests/test_intl/test_json_schema.py ............................................................ [ 11%]
............                                                                                     [ 13%]
tests/test_key_handling/test_key_derivation/test_mnemonic.py ................................... [ 17%]
................................................................................................ [ 28%]
................................................................................................ [ 39%]
................................................................................................ [ 50%]
................................................................................................ [ 61%]
................................................................................................ [ 72%]
................................................................................................ [ 84%]
........................................                                                         [ 88%]
tests/test_key_handling/test_key_derivation/test_path.py ........                                [ 89%]
tests/test_key_handling/test_key_derivation/test_tree.py .......................                 [ 92%]
tests/test_key_handling/test_keystore.py ............                                            [ 93%]
tests/test_utils/test_constants.py ..                                                            [ 94%]
tests/test_utils/test_crypto.py ..........                                                       [ 95%]
tests/test_utils/test_intl.py ..........                                                         [ 96%]
tests/test_utils/test_ssz.py .........                                                           [ 97%]
tests/test_utils/test_validation.py ......................                                       [100%]

=================================== 859 passed in 139.15s (0:02:19) ====================================
```

Are you on Ubuntu 24.04? Maybe it's an OS thing or a preinstalled package I'm missing.

Reply:
Yep I can't reproduce this now either. At least not on Ubuntu 24.04. I think I had an issue on Debian 12 maybe?

OK, I've removed it for now. We can do further tests and bring it back in if necessary

Reply:
Yeah there it is, now our runners fail

```
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    tomli>=1.1.0 from https://files.pythonhosted.org/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl (from mypy==1.11.2->-r requirements_test.txt (line 230))
```

Reply:
Added this back in for now, with a comment as to why it's there. It's weird I agree, it should get picked up from mypy